lib/client.js

```diff
@@ -14,64 +14,101 @@ module.exports = function (dependencies) {
     HTTP2_HEADER_METHOD,
     HTTP2_HEADER_AUTHORITY,
     HTTP2_HEADER_PATH,
-    HTTP2_METHOD_POST
+    HTTP2_METHOD_POST,
+    NGHTTP2_CANCEL,
   } = http2.constants;
 
   function Client (options) {
     this.config = config(options);
     this.healthCheckInterval = setInterval(() => {
-         if (this.session && !this.session.destroyed) {
-             this.session.ping((error, duration) => {
-                 if (error) {
-                     logger("No Ping response after " + duration + " ms");
-                 }
-                 logger("Ping response after " + duration + " ms");
-             })
-         }
-     }, this.config.heartBeat).unref();
+      if (this.session && !this.session.closed && !this.session.destroyed && !this.isDestroyed) {
+        this.session.ping((error, duration) => {
+          if (error) {
+            logger("No Ping response after " + duration + " ms");
```

Comment:
nyc: should return instead of falling through. It may also be useful to log the error message

Reply:
Yes. Make sense. Let's open another PR for this?

Reply:
done